modules/database/src/ioc/db/dbAccessNoThrow.cpp

```diff
@@ -0,0 +1,181 @@
+/*************************************************************************\
+* Copyright (c) 2021 Lucas Russo
+* SPDX-License-Identifier: EPICS
+* EPICS BASE is distributed subject to a Software License Agreement found
+* in file LICENSE that is included with this distribution.
+\*************************************************************************/
+
+#include <stdexcept>
+
+#include "recSup.h"
+#include "dbAccessDefs.h"
+#include "errlog.h"
+#include "dbAccessNoThrow.h"
+
+/*
+ * Maps possible excpetions to DB error codes. Using the
+ * "exception dispatcher" pattern.
+ */
+inline long dbExceptionToLong()
+{
+    try {
+        throw;
+    }
+    catch (std::invalid_argument& e) {
+        errlogPrintf("dbExceptionToLong: %s\n", e.what());
+        return S_db_errArg;
+    }
+    catch (std::bad_alloc& e) {
+        errlogPrintf("dbExceptionToLong: %s\n", e.what());
+        return S_db_noMemory;
+    }
+    catch (std::exception& e) {
+        errlogPrintf("dbExceptionToLong: Unhandled exception\n");
+        /* Return something more meaningful? */
+        return S_db_errArg;
+    }
```

Comment:
I don't think there is much value in translating specific exceptions.

It might be more useful to print the specific exception type `typeid(e).name()` along with `e.what()`.

Reply:
Ok. Is S_db_errArg a good error for this general "unhandled exception" case?